src/transaction.ts

```diff
@@ -245,25 +245,24 @@ export const sendAtomicTransactionComposer = async function (atcSend: AtomicTran
     } as SendAtomicTransactionComposerResults
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
   } catch (e: any) {
+    Config.logger.info('Received error executing Atomic Transaction Composer, for more information enable the debug flag')
     if (Config.debug && typeof e === 'object') {
       e.traces = []
       Config.logger.debug(
-        'Received error executing Atomic Transaction Composer and debug flag enabled; attempting dry run to get more information',
+        'Received error executing Atomic Transaction Composer and debug flag enabled; attempting simulation to get more information',
       )
-      const dryrun = await performAtomicTransactionComposerDryrun(atc, algod)
-
-      for (const txn of dryrun.txns) {
-        if (txn.appCallRejected()) {
+      const simulate = await performAtomicTransactionComposerSimulate(atc, algod)
+      if (simulate.txnGroups[0].failedAt) {
+        for (const txn of simulate.txnGroups[0].txnResults) {
           e.traces.push({
-            trace: txn.appTrace(),
-            cost: txn.cost,
-            logs: txn.logs,
-            messages: txn.appCallMessages,
+            trace: txn.execTrace,
```

Comment:
Previously, we had to use dry run because simulate didn't have traces, I assume the fact we have these fields means that it's up to parity now?

Reply:
@robdmoore based on docstrings/code and documentation on js sdk it seems like they both represent similar notion of traces when txns are app calls. 

See sample ref on simulate execTrace here https://algorand.github.io/js-algorand-sdk/classes/modelsv2.SimulateTransactionResult.html#execTrace

And here is the dryrun definition https://github.com/algorand/js-algorand-sdk/blob/d5ab01407adeba60bf93e315627b2db298de011e/src/dryrun.ts#L335 which seems to be rather similar (i see refs to app calls and logic sig calls)

I am more familiar with simulate though rather than dry run, if you think there is still discrepancy between too we might have to keep dry run too and try to merge them together. But im pretty sure these fields are on par

Reply:
Yeah I think they are finally at parity on tracing, which is great news!